src/regex.js

```diff
@@ -5,12 +5,17 @@ export const testRegex = _.curry((regex, str) => (new RegExp(regex)).test(str))
 export const makeRegex = options => text => RegExp(text, options)
 export const makeAndTest = options => _.flow(makeRegex(options), testRegex)
 
-export const matchAnyWord = _.flow(
+const anyWordToRegexp = _.flow(
   _.words,
-  _.map(makeAndTest('gi')),
-  _.overSome
+  _.map(x => `(?=.*${x})`),
+  _.join('|')
 )
 
+export const matchAnyWord = x => {
+  const regexp = RegExp(anyWordToRegexp(x), 'gi')
+  return y => !!(y && y.match(regexp))
+}
```

Comment:
Since this is almost identical to the implementation of `matchAnyWord`, we should probably factor out the implementation. I'd probably make an internal `matchWords => buildRegex => x => {` method that matchAnyWord and matchAllWords could use.